msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -166,21 +163,39 @@ public Integer getNumberOfDataPoints() {
 
   @Override
   public MsSpectrumType getSpectrumType() {
-    MsSpectrumType spectrumType = SpectrumTypeDetectionAlgorithm.detectSpectrumType(mzValues,
-        intensityValues, getMzBinaryDataInfo().getArrayLength());
-    return spectrumType;
+    if (getCVValue(MzMLCV.cvCentroidSpectrum) != null)
+      return MsSpectrumType.CENTROIDED;
+
+    if (getCVValue(MzMLCV.cvProfileSpectrum) != null)
+      return MsSpectrumType.PROFILE;
+
+    return null;
```

Comment:
If the spectrum type was not specified, do not return null but use the SpectrumTypeDetectionAlgorithm instead